source/material_model/rheology/visco_plastic.cc

```diff
@@ -501,8 +528,11 @@ namespace aspect
 
         if (this->get_parameters().enable_elasticity)
           {
-            for (unsigned int i = 0; i < SymmetricTensor<2,dim>::n_independent_components ; ++i)
-              composition_mask.set(i,false);
+            // Set a mask for the 2*n_independent_components fields representing the viscoelastic
+            // stress tensor components.
+            const std::vector<unsigned int> stress_field_indices = this->introspection().get_indices_for_fields_of_type(CompositionalFieldDescription::stress);
+            for (auto it = stress_field_indices.begin(); it != stress_field_indices.end(); ++it)
+              composition_mask.set(*it, false);
           }
```

Comment:
With #5643 you can simplify this code to:
```suggestion
            const std::vector<unsigned int> &stress_field_indices = this->introspection().get_indices_for_fields_of_type(CompositionalFieldDescription::stress);
            for (unsigned int field_index: stress_field_indices)
              composition_mask.set(field_index, false);
          }
```

But one additional question: The comment says to set a mask representing the viscoelastic stress tensor components, but to me it looks like the code is doing the opposite, it sets a mask that *excludes* all the stress tensor components. Do I misunderstand the code or does the comment need an update?

Reply:
This whole function is actually no longer used since 
#5542 and should be removed.